types/Page.php

```diff
@@ -224,4 +220,35 @@ public function filter(QueryBuilderWhere $add, $tablealias, $colname, $comp, $va
         $pl = $QB->addValue($value);
         $sub->whereOr("$rightalias.title $comp $pl");
     }
+
+    /**
+     * Check if the given id matches at configured namespace (pattern):
+     * simple string or regex pattern with delimiter "/"
+     *
+     * @param string $id
+     * @param string $namespace
+     * @return bool
+     */
+    public function nsMatch($id, $namespace)
+    {
+        $searchNS = getNS($id);
+        if (!$searchNS) {
+            return false; // root
+        }
+
+        // prepare any namespace for preg_match()
+        $searchNS = ':' . $searchNS . ':';
+        // absolute namespace?
+        if (PhpString::substr($namespace, 0, 1) === ':') {
+            $namespace = '^' . $namespace;
+        }
+        // non-regex namespace?
+        if (PhpString::substr($namespace, 0, 1) !== '/') {
+            $namespace = '(?::|^)' . $namespace ;
```

Comment:
if the given namespace is supposed to not be a regex, then it should be escaped using `preg_quote_cb()`

Reply:
All namespaces are now assumed to be regex.